src/core/geometry/qgsgeometry.h

```diff
@@ -390,62 +414,58 @@ class CORE_EXPORT QgsGeometry
      * \param points points describing part to add
      * \param geomType default geometry type to create if no existing geometry
-     * \returns 0 in case of success, 1 if not a multipolygon, 2 if ring is not a valid geometry, 3 if new polygon ring
-     * not disjoint with existing polygons of the feature
+     * \returns OperationResult a result code: success or reason of failure
      */
-    int addPart( const QList<QgsPointXY> &points, QgsWkbTypes::GeometryType geomType = QgsWkbTypes::UnknownGeometry ) SIP_PYNAME( addPoints );
-    // TODO QGIS 3.0 returns an enum instead of a magic constant
+    OperationResult addPart( const QList<QgsPointXY> &points, QgsWkbTypes::GeometryType geomType = QgsWkbTypes::UnknownGeometry ) SIP_PYNAME( addPoints );
 
-    /** Adds a new part to a the geometry.
+    /**
+     * Adds a new part to a the geometry.
      * \param points points describing part to add
      * \param geomType default geometry type to create if no existing geometry
-     * \returns 0 in case of success, 1 if not a multipolygon, 2 if ring is not a valid geometry, 3 if new polygon ring
-     * not disjoint with existing polygons of the feature
+     * \returns OperationResult a result code: success or reason of failure
      */
-    int addPart( const QgsPointSequence &points, QgsWkbTypes::GeometryType geomType = QgsWkbTypes::UnknownGeometry ) SIP_PYNAME( addPointsV2 );
-    // TODO QGIS 3.0 returns an enum instead of a magic constant
+    OperationResult addPart( const QgsPointSequence &points, QgsWkbTypes::GeometryType geomType = QgsWkbTypes::UnknownGeometry ) SIP_PYNAME( addPointsV2 );
 
-    /** Adds a new part to this geometry.
+    /**
+     * Adds a new part to this geometry.
      * \param part part to add (ownership is transferred)
      * \param geomType default geometry type to create if no existing geometry
-     * \returns 0 in case of success, 1 if not a multipolygon, 2 if ring is not a valid geometry, 3 if new polygon ring
-     * not disjoint with existing polygons of the feature
+     * \returns OperationResult a result code: success or reason of failure
      */
-    int addPart( QgsAbstractGeometry *part SIP_TRANSFER, QgsWkbTypes::GeometryType geomType = QgsWkbTypes::UnknownGeometry );
-    // TODO QGIS 3.0 returns an enum instead of a magic constant
+    OperationResult addPart( QgsAbstractGeometry *part SIP_TRANSFER, QgsWkbTypes::GeometryType geomType = QgsWkbTypes::UnknownGeometry );
 
-    /** Adds a new island polygon to a multipolygon feature
+    /**
+     * Adds a new island polygon to a multipolygon feature
      * \param newPart part to add. Ownership is NOT transferred.
-     * \returns 0 in case of success, 1 if not a multipolygon, 2 if ring is not a valid geometry, 3 if new polygon ring
-     * not disjoint with existing polygons of the feature
-     * \note not available in Python bindings
+     * \returns OperationResult a result code: success or reason of failure
```

Comment:
Yup, actually It would be great to have lists of possible return values for all the methods as mentioned by @wonder-sk in the old PR